packages/next/src/template-renderer/index.tsx

```diff
@@ -40,8 +41,17 @@ export async function TemplateRenderer( {
 			>
 				<main>
 					<div className="wp-site-blocks">
-						{ /** TODO: fallback if editorBlocks isn't defined. */ }
-						{ editorBlocks && children( editorBlocks ) }
+						{ editorBlocks ? (
+							children( editorBlocks )
+						) : (
+							<DefaultError
+								error={
+									new Error(
+										'Error: Unable to render content. `editorBlocks` is not defined. This may be due to missing template data or an issue with data fetching.'
+									)
+								}
```

Comment:
Don't remove reset button. Refresh the page on click.

Reply:
Next.js won't let me pass the refresh function from the `TemplateRenderer` to the `DefaultError` component, since the former is a server component and the latter is a client component. I tried creating a middle client component that passed `router.refresh` to `DefaultError`, but it didn't work either.

Thus, I am using `router.refresh` as a fallback for the reset prop if it is not passed.
https://github.com/rtCamp/snapwp/blob/108af92fea33be2bc94b85d1a63c62cdb0adb533/packages/next/src/components/default-error.tsx#L29-L33